gRPC/CmdPassImpl.cc

```diff
@@ -1,5 +1,150 @@
+    File file;
+    file.path = pairs.at(prefix + "file_path");
+    file.filename = pairs.at(prefix + "file_filename");
+    return file;
+}
+
+// Function to convert vector of string pairs to ST2110
+ST2110 stringPairsToST2110(const std::unordered_map<std::string, std::string>& pairs, const std::string& prefix) {
+    ST2110 st2110;
+    st2110.network_interface = pairs.at(prefix + "network_interface");
+    st2110.local_ip = pairs.at(prefix + "local_ip");
+    st2110.remote_ip = pairs.at(prefix + "remote_ip");
+    st2110.transport = pairs.at(prefix + "transport");
+    st2110.remote_port = from_string<int>(pairs.at(prefix + "remote_port"));
+    st2110.payload_type = from_string<int>(pairs.at(prefix + "st_payload_type"));
+    return st2110;
+}
+
+// Function to convert vector of string pairs to MCM
+MCM stringPairsToMCM(const std::unordered_map<std::string, std::string>& pairs, const std::string& prefix) {
+    MCM mcm;
+    mcm.conn_type = pairs.at(prefix + "conn_type");
+    mcm.transport = pairs.at(prefix + "transport");
+    mcm.transport_pixel_format = pairs.at(prefix + "transport_pixel_format");
+    mcm.ip = pairs.at(prefix + "ip");
+    mcm.port = from_string<int>(pairs.at(prefix + "port"));
+    mcm.urn = pairs.at(prefix + "urn");
+    return mcm;
+}
+
+// Function to convert vector of string pairs to Payload
+Payload stringPairsToPayload(const std::unordered_map<std::string, std::string>& pairs, const std::string& prefix) {
+    Payload payload;
+    payload.type = static_cast<payload_type>(from_string<int>(pairs.at(prefix + "payload_type")));
+    if (payload.type == video) {
+        payload.video = stringPairsToVideo(pairs, prefix);
+    } else if (payload.type == audio) {
+        payload.audio = stringPairsToAudio(pairs, prefix);
+    }
+    return payload;
+}
+
+// Function to convert vector of string pairs to StreamType
+StreamType stringPairsToStreamType(const std::unordered_map<std::string, std::string>& pairs, const std::string& prefix) {
+    StreamType streamType;
+    streamType.type = static_cast<stream_type>(from_string<int>(pairs.at(prefix + "stream_type")));
+    if (streamType.type == file) {
+        streamType.file = stringPairsToFile(pairs, prefix);
+    } else if (streamType.type == st2110) {
+        streamType.st2110 = stringPairsToST2110(pairs, prefix);
+    } else if (streamType.type == mcm) {
+        streamType.mcm = stringPairsToMCM(pairs, prefix);
+    }
+    return streamType;
+}
+
+// Function to convert vector of string pairs to Stream
+Stream stringPairsToStream(const std::unordered_map<std::string, std::string>& pairs, const std::string& prefix) {
+    Stream stream;
+    stream.payload = stringPairsToPayload(pairs, prefix);
+    stream.stream_type = stringPairsToStreamType(pairs, prefix);
+    return stream;
+}
+
+// Function to convert vector of string pairs to Config
+static Config stringPairsToConfig(const std::vector<std::pair<std::string, std::string>>& pairs) {
```

Comment:
let's make the other functions as static to not expose not needed symbols.

Reply:
Fixed in https://github.com/OpenVisualCloud/Intel-Tiber-Broadcast-Suite/pull/49/commits/a25878f7b55f8de4ffe4b4d7121ad3707906d3c8